### DISCUSSION TITLE
Spawn Redis resources in a separated NameSpace ?

### DISCUSSION THREAD

**Discussion by xakraz** [OT-CONTAINER-KIT/redis-operator#87] [Ideas]
Hi there !

I have a question / discussion about where to deploy the underlying resources for Redis managed by this operator.

Currently, all of our MicroServices are using a dedicated Redis instance and we are using RedisLabs managed services.
For performances and cost reasons, we want to run the Redis instances on the same cluster than the one used by our MicroServices.

All our MicroServices are deployed in the same shared namespace. This is not ideal and seems to not follow MicroServices best practices on Kubernertes but it works for now and is convenient.

In the process of thinking how we could integrate `OT-CONTAINER-KIT/redis-operator` in our stacks, I was wondering if it would be possible and a good idea to:
* Have the Redis manifest among our MS manifests
* BUT have the underling Redis resources deployed in a dedicated namespace.

This will have cons for sure, regarding all the "isolations" namespace can provide (RBAC, Quotas, ...) but in a very naive way, it would look like having a "remote managed service" like we could have by using RedisLabs, Aws ElastiCache or GCP Redis but still running in the same cluster.

Any ideas, any thoughts about such organization ?

Thanks 🙏

**Comment by iamabhishek-dubey** [comment]
Feature is already added

**Reply by xakraz** [reply]
Cool! How is it working?

I am expecting:
* 1 `Redis` object from the `redis.redis.opstreelabs.in/v1beta1` CRD among our app manifests (Therefore deployed in the same namespace as the MicroService)
* But `Pods`, `Services` and `ServicesMonitors` deployed in another namespace

Then having my micro-service accessing the Redis with `<REDIS-NAME>.redis`
